File: src/main/java/ru/rsavin/socialnetwork/web/controller/RestExceptionHandler.java
package ru.rsavin.socialnetwork.web.controller;

import org.springframework.dao.EmptyResultDataAccessException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.context.request.WebRequest;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@RestControllerAdvice
public class RestExceptionHandler extends ResponseEntityExceptionHandler {
    @ExceptionHandler({ EmptyResultDataAccessException.class })
    public ResponseEntity<Object> handleDataNotFoundException(Exception ex, WebRequest request) {
        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
    }
}

File: app/src/main/kotlin/ru/rsavin/socialnetwork/web/controller/RestExceptionHandler.kt
package ru.rsavin.socialnetwork.web.controller

import org.springframework.dao.EmptyResultDataAccessException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.context.request.WebRequest

@RestControllerAdvice
class RestExceptionHandler {
    @ExceptionHandler(EmptyResultDataAccessException::class)
    fun handleDataNotFoundException(ex: Exception?, request: WebRequest?): ResponseEntity<Any?> {
        return ResponseEntity(HttpStatus.NOT_FOUND)
    }
}